Add LiteLLM integration documentation (#45740)
home-assistant/home-assistant.github.io: source/_integrations/litellm.markdown

---
title: LiteLLM
description: Instructions on how to integrate LiteLLM as a conversation agent
ha_category:
  - AI
ha_release: 2026.7

> ha_iot_class: Cloud Polling

Reviewer: ```suggestion
ha_iot_class: Local Polling
```
In core, it says this is local polling. Which one is correct?
https://github.com/home-assistant/core/pull/172960/changes#diff-064351471f6e6742d74b077fc19acbd0f1a771b324608a7238f70d16eeb43114R10

Author: Good point, I believe that it should be Cloud pooling (have changed in Core as well now). Even tho we can have LiteLLM self-hosted, and in theory LiteLLM can serve local models, it's most common usage is to be a proxy to several OpenAI, etc endpoints, hence internet connection is required in the end.

ha_config_flow: true
ha_codeowners:
  - '@luismalves'
ha_domain: litellm
ha_integration_type: service
ha_platforms:
  - conversation
ha_quality_scale: bronze
---

The [LiteLLM](https://www.litellm.ai/) {% term integration %} allows you to use a LiteLLM proxy as a conversation agent in Home Assistant.

LiteLLM exposes a single OpenAI-compatible API in front of a wide range of model providers, so you can use almost any model through one endpoint. This is similar to a service like OpenRouter, with the added benefit that you can run LiteLLM yourself if you prefer to keep everything under your own control. This integration lets you point Home Assistant at any LiteLLM proxy, whether it is a hosted instance or one you self-host.

{% include integrations/config_flow.md %}

{% configuration_basic %}
URL:
  description: The base URL of your LiteLLM proxy, including the host and port. The integration automatically appends the OpenAI `/v1` path if it is missing.
API key:
  description: An optional LiteLLM API key or virtual key. Leave it empty if your proxy does not require authentication.
{% endconfiguration_basic %}

> ## Prerequisites

Reviewer: are these prereqs for adding the integration or for adding a conversation agent?

Author: Thanks for point it out. Those were meant to be for the actual integration and not the conv. agent

You need access to a [LiteLLM proxy](https://docs.litellm.ai/docs/simple_proxy) that is reachable from your Home Assistant instance, with at least one model configured. You can use a hosted LiteLLM proxy or [run one yourself](https://docs.litellm.ai/docs/simple_proxy). The integration discovers the available models from the proxy during setup.

## Supported functionality

### Conversation agent

> The integration adds a conversation agent for each configured model that can be used in [Assist](/voice_control/). When you grant it control of Home Assistant, the agent can call the configured tools to interact with your devices and entities. To choose which devices and entities the agent can access, see [exposing entities to Assist](/voice_control/voice_remote_expose_devices/).

Reviewer: ```suggestion

### Adding a conversation agent

1. Go to {% my integrations title="**Settings** > **Devices & services**" %} and select your LiteLLM entry.
2. Select **Add conversation agent**.
3. Fill in the fields:

{% configuration_basic %}
Model:
   description: "The model to use for the conversation agent. The list is discovered from your LiteLLM proxy."
Instructions:
   description: "Instruct how the LLM should respond. This field supports [templates](/docs/configuration/templating/)."
Control Home Assistant:
   description: "Select which tools the agent can use to interact with your devices and entities. Leave empty if you do not want the agent to control Home Assistant."
{% endconfiguration_basic %}

When you grant the agent control of Home Assistant, it can call the configured tools to interact with your devices and entities. To choose which devices and entities the agent can access, see [exposing entities to Assist](/voice_control/voice_remote_expose_devices/).

The agent can be used in [Assist](/voice_control/) like any other conversation agent.

To change the model, instructions, or tools later, select the conversation agent and choose **Reconfigure conversation agent**.

```
Would it be worth adding the steps?

---
title: LiteLLM
description: Instructions on how to integrate LiteLLM as a conversation agent
ha_category:
  - AI
  - Voice
ha_release: 2026.7
ha_iot_class: Local Polling
ha_config_flow: true
ha_codeowners:
  - '@luismalves'
ha_domain: litellm
ha_integration_type: service
ha_platforms:
  - conversation
ha_quality_scale: bronze
---

The [LiteLLM](https://www.litellm.ai/) {% term integration %} allows you to use a self-hosted LiteLLM proxy as a conversation agent in Home Assistant.

LiteLLM exposes an OpenAI-compatible API in front of a wide range of model providers. Because the proxy is self-hosted, this integration lets you point Home Assistant at your own endpoint instead of a hosted service.

{% include integrations/config_flow.md %}

{% configuration_basic %}
URL:

>   description: The base URL of your LiteLLM proxy, including the host and port (for example `http://homeassistant.local:4000`). The integration automatically appends the OpenAI `/v1` path if it is missing.

Author: Fixed — the Prerequisites paragraph no longer references specific endpoint paths; it now simply states the integration discovers the available models from the proxy during setup, removing the `/v1` inconsistency.

API key:
  description: An optional LiteLLM API key or virtual key. Leave it empty if your proxy does not require authentication.
{% endconfiguration_basic %}

## Prerequisites

> You need a running [LiteLLM proxy](https://docs.litellm.ai/docs/simple_proxy) that is reachable from your Home Assistant instance, with at least one model configured. The proxy exposes the available models through its `/model/info` (or `/models`) endpoint, which the integration uses to populate the model selection.

Author: Fixed — see above; the endpoint paths were removed from the Prerequisites paragraph.

## Supported functionality

### Conversation agent

The integration adds a conversation agent for each configured model that can be used in [Assist](/voice_control/). When you grant it control of Home Assistant, the agent can call the configured tools to interact with your devices and entities.

## Removing the integration

> This integration follows standard integration removal, no extra steps are required.

Author: This is the standard, verbatim removal boilerplate used across all Home Assistant integration documentation pages (e.g. open_router, ollama). Kept as-is for consistency with the rest of the docs.

---
title: LiteLLM
description: Instructions on how to integrate a LiteLLM proxy as a conversation agent in Home Assistant
ha_category:
  - AI

> ha_release: 2026.7

Reviewer 2: ```suggestion
ha_release: 2026.8
```

{% include integrations/config_flow.md %}

{% configuration_basic %}
URL:
  description: The base URL of your LiteLLM proxy, including the host and port. The integration automatically appends the OpenAI `/v1` path if it is missing.
API key:
  description: An optional LiteLLM API key or virtual key. Leave it empty if your proxy does not require authentication.
{% endconfiguration_basic %}

## Prerequisites

Before you add this integration, you need the following:

- **A LiteLLM proxy that Home Assistant can reach.** This is the endpoint the integration connects to. You can use a hosted LiteLLM proxy or [run one yourself](https://docs.litellm.ai/docs/simple_proxy).
- **At least one model configured on the proxy.** The integration discovers the available models from the proxy during setup and creates a conversation agent for each one.
- **An API key, if your proxy requires authentication.** This is used to authorize Home Assistant's requests to the proxy. You can leave it empty if your proxy does not require one.

## Supported functionality

> 

Reviewer: ```suggestion
```
I think in this case, we could even skip this title and have adding the conversation agent as part of the setup section.

Author: Looking good now in my opinion, with this change 👍

### Adding a conversation agent

1. Go to {% my integrations title="**Settings** > **Devices & services**" %} and select your LiteLLM entry.
2. Select **Add conversation agent**.
3. Fill in the fields:

{% configuration_basic %}
Model:
   description: "The model to use for the conversation agent. The list is discovered from your LiteLLM proxy."
Instructions:
   description: "Instruct how the LLM should respond. This field supports [templates](/docs/configuration/templating/)."
Control Home Assistant:
   description: "Select which tools the agent can use to interact with your devices and entities. Leave empty if you do not want the agent to control Home Assistant."
{% endconfiguration_basic %}

When you grant the agent control of Home Assistant, it can call the configured tools to interact with your devices and entities. To choose which devices and entities the agent can access, see [exposing entities to Assist](/voice_control/voice_remote_expose_devices/).

The agent can be used in [Assist](/voice_control/) like any other conversation agent.

To change the model, instructions, or tools later, select the conversation agent and choose **Reconfigure conversation agent**.

> 

Reviewer: ```suggestion
```
